_studio/mfx_lib/shared/src/mfx_brc_common.cpp

```diff
@@ -153,14 +153,21 @@ mfxStatus cBRCParams::Init(mfxVideoParam* par, bool fieldMode)
 
     mfxExtCodingOption * pExtCO = (mfxExtCodingOption*)Hevc_GetExtBuffer(par->ExtParam, par->NumExtParam, MFX_EXTBUFF_CODING_OPTION);
 
-    bHRDConformance =  (pExtCO && pExtCO->NalHrdConformance == MFX_CODINGOPTION_OFF) ? 0 : 1;
+    HRDConformance = MFX_BRC_NO_HRD;
+    if (pExtCO)
+    {
+        if (!IsOff(pExtCO->NalHrdConformance) && !IsOff(pExtCO->VuiNalHrdParameters))
+            HRDConformance = MFX_BRC_HRD_STRONG;
+        else if (IsOn(pExtCO->NalHrdConformance) && IsOff(pExtCO->VuiNalHrdParameters))
+            HRDConformance = MFX_BRC_HRD_WEAK;
+    }
 
-    if (bHRDConformance)
+    if (HRDConformance)
```

Comment:
Better to use explicit comparison with NO_HRD enum and not rely on defaults. Same applicable to many places below
```
if (HRDConformance != MFX_BRC_NO_HRD)
```

Reply:
changed